### DISCUSSION TITLE
How to add additional fields to an existing non-empty json object ?

### DISCUSSION THREAD

**Discussion by GokulK-123** [schibsted/jslt#240] [General]
I am trying to write a function using JSLT that receives a non-empty json object, obj1, as one of the function parameter. I need to add a couple of fields to it and return it back to the caller. But this doesn't seem to be working. How can I add additional fields to an existing non-empty json object and return it back ?

def func1 (obj1, resp){
obj1.prop1 = resp.prop1
obj1.prop2 = resp.prop2
$obj1
}

def main(){
"data": func1(obj1,resp)
}

**Comment by larsga** [comment]
The easiest way is to simply plus the two objects together:

```
def func1 (obj1, resp) 
  $obj1 + {
    "prop1" : $resp.prop1,
    "prop2" : $resp.prop2
  }
```
